libs/ledger-live-common/src/exchange/providers/swap.ts

```diff
@@ -213,6 +225,34 @@ export const getProvidersData = async () => {
   return transformData(providersData.data);
 };
 
+/**
+ * Retrieves the currency data for a given ID
+ * coins and tokens are litteraly the same on our case but we need to call the correct endpoint.
+ * @param currencyId The unique identifier for the currency.
+ * @param type Specifies the type of currency, either "tokens" or "coins".
+ * @returns A promise that resolves to the currency data including ID, serialized config, and signature.
+ */
+export const findExchangeCurrencyData = async (
+  currencyId: string,
+  type: "tokens" | "coins",
+): Promise<CurrencyData> => {
+  const currencyData = (await network({
+    url:
+      `https://crypto-assets-service.api.ledger.com/v1/${type}` +
+      "?output=id,exchange_app_config_serialized,exchange_app_signature" +
+      `&id=${currencyId}`,
+  })) as { data: CurrencyDataResponse };
+  // throw errors/warning if many currencies fetched with the same ID, can it happen?
```

Comment:
I notice that this comment doesn't do what it says it will do and throw a warning if length > 1. Do we want to? 

And out of interest, do we know why the response is an array if only one object is possible? (E.g. can this endpoint can take `currencyId: string[]`, too? Or should it really _not_ be an array in response?)

Reply:
Yeah this comment is more a TODO: xxx I'll update it. 
And this point is used to returned all currencies and we can filter them based on chains, ids ... so array is required. It is just by filtering with IDs we assume only one result but I remember that on some edge cases (renaming and other) it could happen ids are not unique